chromadb/api/models/Collection.py

```diff
@@ -478,11 +487,23 @@ def upsert(
             ids, embeddings, metadatas, documents, images, uris
         )
 
+        # We need to compute the embeddings if they're not provided
         if embeddings is None:
+            # At this point, we know that one of documents or images are provided from the validation above
             if documents is not None:
                 embeddings = self._embed(input=documents)
-            else:
+            elif images is not None:
                 embeddings = self._embed(input=images)
+            else:
+                if uris is None:
+                    raise ValueError(
```

Comment:
Think this case is caught by the validator, so I think you only need the data loader error here